tests_and_analysis/test/euphonic_test/test_qpoint_frequencies.py

```diff
@@ -528,7 +529,7 @@ def test_calculate_dos_with_0_inv_cm_bin_doesnt_raise_runtime_warn(self):
         qpt_freqs = get_qpt_freqs(
             'quartz', 'quartz_666_qpoint_frequencies.json')
         ebins = np.arange(0, 1300, 4)*ureg('1/cm')
-        with pytest.warns(None) as warn_record:
+        with warnings.catch_warnings(record=True) as warn_record:
```

Comment:
Pytest docs suggest an alternative way of promoting warnings to errors: https://docs.pytest.org/en/7.1.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

```python
with warnings.catch_warnings():
    warnings.simplefilter("error")
    ...
```

Not clear if our way of doing it gives better information, probably not. (I think it will just tell us the number of warning but not what they were?)

Reply:
Yeah, there seems to be a bit of a disagreement on what is the best way to do things, this replacement is the most like-for-like replacement, even though it's not what's listed in the docs. See https://github.com/pytest-dev/pytest issue 9404 and its first comment, this is also the drop in replacement implemented by Scipy, see https://github.com/scipy/scipy PR 15192. (not linking directly because I don't want to Spam their issues/PRs).

I guess where we assert that there is 1 warning we could also check the type of warning

Reply:
With `pytest.warns`, as well as providing a warning subtype you can provide a pattern to match in the warning message. So where we _require_  a warning that's the tool for the job.

Reply:
Good point, I don't know why those were written with `None` in the first place...